### PR TITLE
[SPARK-41067][CONNECT][PYTHON] Implement `DataFrame.stat.cov`

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
Implement `DataFrame.stat.cov` with a proto message

Implement `DataFrame.stat.cov` for scala API
Implement `DataFrame.stat.cov` for python API

### Why are the changes needed?
for Connect API coverage

### Does this PR introduce _any_ user-facing change?
'No'. New API

### How was this patch tested?
New test cases.

### CODE REVIEW

**Comment by beliefer** [comment]
ping @HyukjinKwon @zhengruifeng @grundprinzip @amaliujia

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala [hunk 1]
```diff
@@ -328,6 +329,16 @@ class SparkConnectPlanner(session: SparkSession) {
       .logicalPlan
   }
 
+  private def transformStatCov(rel: proto.StatCov): LogicalPlan = {
```
Comment (grundprinzip) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:332]: One of the reasons we haven't implemented that until now is that this performs the execution of the query directly as part of the logical plan.

If you compare this to the other methods they all invoke the DF transformation directly.

One question would be to identify how to rewrite the `cov()` function to be a DF transformation or use it as such.

Comment (zhengruifeng) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:332]: `df.stat.cov` returns a double value, which always trigger a job to compute it.

the users will need to wait for this computation.

**Review by zhengruifeng — APPROVED** [review]

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala (hunk 1 above)
Comment (beliefer) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:343]: Yes. So we can collect the result of execution and wraps it with `LocalRelation`.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by beliefer** [comment]
@HyukjinKwon @zhengruifeng @grundprinzip Thank you

**Pull request closed without merging**